src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int SIZE_DIVIDER = 2;
+    private static final int SIZE_SHIFTER = 1;
+    private static final int ZERO_SIZE = 0;
+    private static final int MINUS_ONE = -1;
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, last, null);
+        if (first == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexElement(index);
+        if (index == size) {
+            add(value);
+        } else {
+            linkPreviousElement(value, getNodeByIndex(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexPosition(index);
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexPosition(index);
+        Node<T> tempNode = getNodeByIndex(index);
+        T oldValue = tempNode.value;
+        tempNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexPosition(index);
+        return unlinkNode(getNodeByIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> current = first; current != null; current = current.next) {
+            if (object == null && current.value == null) {
+                unlinkNode(current);
+                return true;
+            } else if (object != null && object.equals(current.value)) {
+                unlinkNode(current);
+                return true;
+            }
```

Comment:
body is the same in both cases, merge conditions together
look up code for Objects.equals() and write the same here
a == b || a != null && a.equals(b)